### PR TITLE
[@mantine/hooks] use-previous: Implement hook

### PR DESCRIPTION
**Pull request by kabeer05**

### CODE REVIEW

**Comment by rtivital** [comment]
@kabeer05 Please add hook documentation

**Comment by kabeer05** [comment]
> @kabeer05 Please add hook documentation

hook documentation?

**Comment by rtivital** [comment]
Yes, each hook must have documentation that is available on mantine.dev, see docs folder.

File: docs/src/docs/hooks/use-previous.mdx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+- `previousValue`: The previous value of the state.
+
+## Definition
+
+```tsx
+function usePrevious(
+  value: any
+): {
+  previousValue: any
```
Comment (rtivital) [inline docs/src/docs/hooks/use-previous.mdx:40]: Definition is incorrect, hook returns value, not object, it should be `function usePrevious<T>(value: T): T`

File: docs/src/docs/hooks/use-previous.mdx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+source: 'mantine-hooks/src/use-previous/use-previous.ts'
+---
+
+import { HooksDemos } from '@mantine/demos';
+
+## Usage
+
+`use-previous` hook stores the previous value of a state in a ref. Returns undefined on initial render and the previous value of a state after rerender.
+
+## API
```
Comment (rtivital) [inline docs/src/docs/hooks/use-previous.mdx:20]: Remove API section, it is already described well in Usage section

File: docs/src/docs/hooks/use-previous.mdx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+title: 'use-previous'
+order: 1
+slug: /hooks/use-previous/
+description: 'Get the previous value of a state'
+import: "import { usePrevious } from '@mantine/hooks';"
+docs: 'hooks/use-previous.mdx'
+source: 'mantine-hooks/src/use-previous/use-previous.ts'
+---
+
+import { HooksDemos } from '@mantine/demos';
```
Comment (rtivital) [inline docs/src/docs/hooks/use-previous.mdx:14]: Remove this import

**Review by rtivital — CHANGES_REQUESTED** [review]

File: docs/src/docs/hooks/use-previous.mdx (hunk 1 above)
Comment (kabeer05) [reply docs/src/docs/hooks/use-previous.mdx:40]: oh mbmb

**Comment by rtivital** [comment]
@kabeer05 why did you close the PR?

**Comment by kabeer05** [comment]
> @kabeer05 why did you close the PR?

i will create a new one was having problems with this one

**Pull request closed without merging**